lib/babel.js

```diff
@@ -74,8 +77,33 @@ if (inputSourceMap) {
 	options.inputSourceMap = JSON.parse(inputSourceMap.map);
 }
 
+// load up custom compilers
+process.argv.slice(3).forEach(function (arg) {
+	var match = /^--compilers=.+?:(.+)$/i.exec(arg);
+	if (match) {
+		require(match[1]);
+	}
+});
+
 // include test file
-var transpiled = babel.transformFileSync(testPath, options);
+var Module = module.constructor;
+var source = fs.readFileSync(testPath, 'utf8');
+var extension = path.extname(testPath);
+
+var compiler = Module._extensions[extension];
+if (compiler) {
+	compiler(
+		{
+			_compile: function (s, f) {
+				source = s;
+				testPath = f;
```

Comment:
Throw if any of these are empty? This isn't a documented Node API so people probably copy/paste from other require hooks and will dutifully pass the same filename but if they don't it may be quite tricky to debug.